docs/DeveloperGuide.adoc

```diff
@@ -122,125 +119,647 @@ image::ModelClassDiagram.png[]
+* `*find*` - Finds a task in Mod Manager by its description.
+* `*upcoming*` - Finds upcoming tasks (for tasks with a specified time period) in Mod Manager.
+* `*search*`- Searches for tasks that occur on your specified date, month, or year in Mod Manager.
+* `*edit*` - Edits the information of a task in Mod Manager.
+* `*delete*` - Deletes a task from the `Module` and Mod Manager.
+
+==== Implementation Details
+
+===== Adding a task
+// to extend on Task, ScheduledTask and NonScheduledTask
+The add task feature allows users to add a task to Mod Manager.
+This feature is facilitated by `TaskCommandParser`, `TaskAddCommandParser` and `
+TaskAddCommand`.
+The operation is exposed in the `Model` interface as `Model#addTask()`.
+
+Given below is an example usage scenario and how the `*task* add` mechanism behaves at each step:
+
+1. The user executes the `*task* add` command and provides the module code, the description of the task (both compulsory),
+and a time period (optional), which consists of a date (for example, `15/04/2020`) or a date and time (`15/04/2020` and `23:59`) of the task to be added.
+2. `TaskAddCommandParser` creates a new `Task` based on the module code, description, and time period (if provided).
+3. `TaskAddCommandParser` creates a new `TaskAddCommand` based on the task.
+4. `LogicManager` executes the `TaskAddCommand`.
+5. `ModManager` adds the task to the `UniqueTaskList`.
+6. `ModelManager` updates the `filteredTasks` in `ModelManager`.
+
+The following sequence diagram shows how the `*task* add` command works:
+
+.Sequence Diagram for `task add` Command
+image::TaskAddSequenceDiagram.png[]
+
+NOTE: The lifeline for `TaskCommandParser`, `TaskAddCommandParser` and `TaskAddCommand` should end at
+the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+The following activity diagram summarizes what happens when a user executes a `*task* add` command:
+
+.Activity Diagram for `task add` Command
+image::TaskAddActivityDiagram.png[]
+
+===== Search tasks by date
+The search task feature allows users to search all tasks that occur on the specified date, month, or year.
+This feature is facilitated by `TaskCommandParser`, `TaskSearchCommandParser` and `
+TaskSearchCommand`.
+The operation is exposed in the `Model` interface as `Model#updateFilteredTaskList()`.
+
+Given below is an example usage scenario and how the `*task* search` mechanism behaves at each step:
+
+1. The user executes the `*task* search` command and provides the day, month, or year, or any combination of which
+that they want to search for search for.
+2. `TaskSearchCommandParser` creates a new `TaskSearchCommand` based on the names.
+3. `LogicManager` executes the `TaskSearchCommand`.
+4. `ModelManager` updates the `filteredTasks` in `ModelManager`.
+
+The following sequence diagram shows how the `*task* search` command works:
+
+.Sequence Diagram for `*task* search` Command
+image::TaskSearchSequenceDiagram.png[]
```

Comment:
TaskSearchPredicate is rather hard to see due to the white font, would be good to change the colour